compiler/ksp_compiler_extras.py

```diff
@@ -487,34 +527,43 @@ def visitFunctionCall(self, parent, node, *args):
             # visit all children except the first one (the key-id)
             for child in node.get_childnodes()[1:]:
                 self.dispatch(node, child, *args)
+
             return False
 
     def visitFunctionDef(self, parent, node, *args):
         if node.name.identifier in user_defined_functions:
             raise ParseException(node, 'A variable or a function with the same name already exists!')
+
         user_defined_functions[node.name.identifier] = node
+
         return True
 
     def visitDeclareStmt(self, parent, node, *args):
         name = str(node.variable)
         is_ui_control = [x for x in node.modifiers if x.startswith('ui_')]
+        ui_controls_without_params = \
```

Comment:
This would be better to have in `ksp_builtins_data.py`, and then made into a set by `ksp_builtins.py`.

Reply:
In case this set needs to be reused in the future some where else. Or if we add version targeting and some of these controls do not exist in that version. Also makes sense to have all the inbuilts in one place, as we've done with `functions_with_forced_parentheses`, so that its easier to adjust in the future.

Reply:
OK in that case I think it's better to add a generic array that lists ALL UI control types along with the number of arguments they have. probably a dict then, not a set.

Reply:
Sounds good 👍